tools/snapshot/node_mksnapshot.cc

```diff
@@ -11,48 +11,103 @@
 #include "util-inl.h"
 #include "v8.h"
 
+int BuildSnapshot(int argc, char* argv[]);
+
 #ifdef _WIN32
 #include <windows.h>
 
-int wmain(int argc, wchar_t* argv[]) {
+int wmain(int argc, wchar_t* wargv[]) {
+  // Windows needs conversion from wchar_t to char.
+
+  // Convert argv to UTF8.
+  char** argv = new char*[argc + 1];
+  for (int i = 0; i < argc; i++) {
+    // Compute the size of the required buffer
+    DWORD size = WideCharToMultiByte(CP_UTF8,
+                                     0,
+                                     wargv[i],
+                                     -1,
+                                     nullptr,
+                                     0,
+                                     nullptr,
+                                     nullptr);
+    if (size == 0) {
+      // This should never happen.
+      fprintf(stderr, "Could not convert arguments to utf8.");
+      exit(1);
+    }
```

Comment:
This is copied from node_main.cc (copied instead of reused since node_main.cc is supposed to only include node.h and it does not seem to be the right time to add these into node.h yet). I'd prefer to leave this the way it is for this patch and maybe change it in a follow-up, WDYT?

Reply:
Ah, I didn't realize that this was copied, we can do it in a separate PR then. :)